# Kruskal

In [20]:
from scipy import stats
import statistics 
import pandas as pd
from pingouin import cochran
import numpy as np

## Prepare the data

In [2]:
dekat = [59,110,132,143,165]
sedang = [77,99,128,144]
jauh = [89,102,121]

In [3]:
model_kruskal = stats.kruskal(dekat,sedang, jauh)
print(model_kruskal)

KruskalResult(statistic=np.float64(1.0038461538461547), pvalue=np.float64(0.6053653754225883))


![data](./assets/kurskal.png)

Karena nilai _pvalue_ adalah 0.6053 dimana lebih besar dari 0.05, maka hipotesa H0 diterima, dimana tidak ada perbedaan antara jarak jauh dengan keterlambatan.

## Sources :

* [https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.kruskal.html](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.kruskal.html)

# Median (alexandergovern)

In [4]:
wilayah_A = [4.5,5,4,6,4.8,5.2,5.5,5.8,4.3,6.2]
wilayah_B = [6,6.5,7,5.5,6.2,6.8,7.1,6.9,5.8,6.4]
wilayah_C = [3.5,3.8,4,3.9,4.2,3.7,4.1,4.3,4.5,4.4]

In [6]:
model_aleandergovern = stats.alexandergovern(wilayah_A,wilayah_B,wilayah_C)
print(model_aleandergovern)

AlexanderGovernResult(statistic=np.float64(39.2347770629932), pvalue=np.float64(3.02187441312258e-09))


# Chisquare Independen

In [28]:
import numpy as np

# Pendidikan (ordinal)
pendidikan = np.array([
    2, 3, 1, 2, 1,
    3, 2, 3, 1, 2,
    3, 2, 1, 3, 2
])

# Status pekerjaan (biner)
status_pekerjaan = np.array([
    1, 1, 0, 1, 0,
    1, 0, 1, 0, 1,
    1, 0, 0, 1, 1
])

In [31]:
df = pd.DataFrame({
    "Pendidikan": pendidikan,
    "Status": status_pekerjaan
})
df

Pendidikan  Status
0            2       1
1            3       1
2            1       0
3            2       1
4            1       0
5            3       1
6            2       0
7            3       1
8            1       0
9            2       1
10           3       1
11           2       0
12           1       0
13           3       1
14           2       1

In [30]:
# Tabel kontingensi
kontingensi = pd.crosstab(df["Pendidikan"], df["Status"])
print(kontingensi)

model_csquare = stats.contingency.chi2_contingency(kontingensi)
model_csquare

Status      0  1
Pendidikan      
1           4  0
2           2  4
3           0  5


Chi2ContingencyResult(statistic=np.float64(9.444444444444445), pvalue=np.float64(0.008895389037043957), dof=2, expected_freq=array([[1.6, 2.4],
       [2.4, 3.6],
       [2. , 3. ]]))

# Cochran

In [16]:
obat_A = [1, 0, 1, 0, 1, 1, 0, 1, 0, 1]
obat_B = [1, 1, 1, 1, 0, 1, 1, 1, 1, 0]
obat_C = [0, 1, 0, 1, 0, 0, 1, 0, 1, 1]

df = pd.DataFrame({'obatA':obat_A,'obatB':obat_B,'obatC':obat_C})
df

obatA  obatB  obatC
0      1      1      0
1      0      1      1
2      1      1      0
3      0      1      1
4      1      0      0
5      1      1      0
6      0      1      1
7      1      1      0
8      0      1      1
9      1      0      1

In [18]:
model_cochran = cochran(df)
print(model_cochran)

         Source  dof    Q     p-unc
cochran  Within    2  1.4  0.496585


Karena nilai p-unv (p value) atau signifikan 0.497, maka H0 diterima, yg mana tidak ada perbedaan antara obat A, B dan C

# Chiquare

In [21]:
data = np.array([
    [15, 25, 10],  # Remaja (17–25) [Brand A, Brand B, Brand C]
    [20, 15, 15],  # Dewasa Muda (26–35) [Brand A, Brand B, Brand C]abs
    [10, 20, 20]   # Dewasa (36–50) [Brand A, Brand B, Brand C]
])

In [22]:
chi2, p, dof, expected = stats.contingency.chi2_contingency(data)

print("Chi-Square:", chi2)
print("p-value:", p)
print("Derajat bebas:", dof)
print("Frekuensi harapan:\n", expected)

Chi-Square: 9.166666666666668
p-value: 0.05706596704573361
Derajat bebas: 4
Frekuensi harapan:
 [[15. 20. 15.]
 [15. 20. 15.]
 [15. 20. 15.]]


# Friedman

In [23]:
direktif = [3, 2, 1, 3, 2, 3, 1, 1, 2, 2, 3, 1, 2, 3, 3]
supportif = [1, 1, 2, 2, 1, 1, 2, 3, 1, 1, 1, 2, 1, 2, 1]
partisipatif = [2, 3, 3, 1, 3, 2, 3, 2, 3, 3, 2, 3, 3, 1, 2]

In [24]:
mode_fredman = stats.friedmanchisquare(direktif,supportif,partisipatif)
print(mode_fredman)

FriedmanchisquareResult(statistic=np.float64(6.933333333333337), pvalue=np.float64(0.031220927161230847))


Maka menolak HO